### PR TITLE
WIP - Diva reporting

### PR DESCRIPTION
**Pull request by akihikot**

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Can one of the admins verify this patch?

**Comment by akihikot** [comment]
@mrizzi @PhilipCattanach Here are instructions for generating a report for DayTrader application.  It would be nice if you can check and review the results within this PR. 

1. Build windup 
```
$ JAVA_HOME=<path to java8 sdk> mvn clean install -DskipTests
```
2. Prepare DayTrader application code at the sibling directory of windup.
```
$ cd .. & git clone https://github.com/WASdev/sample.daytrader7.git
```
3. Comment out `@Ignore` on DivaTest.testDayTrader() 
```
$ vi rules-java-diva/tests/src/test/java/org/jboss/windup/rules/apps/diva/DivaTest.java
```
4. Run the test and view the generated html.
```
$ JAVA_HOME=<path to java8 sdk> mvn clean test -f rules-java-diva/tests -Dtest=DivaTest#testDayTrader
$ open rules-java-diva/tests/target/Windup/*/reports/divareport_sample_daytrader7.html
```

File: rules-java-diva/tests/src/test/java/org/jboss/windup/rules/apps/diva/DivaTest.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                    .has("sql", P.without("BEGIN", "COMMIT", "ROLLBACK")).toList();
+            Assert.equals(ops.size(), 1);
+        }
+    }
+
+    @Ignore
+    @Test
+    public void testDayTrader() throws IOException {
+        try (GraphContext context = factory.create(true)) {
+            Path inputPath = Paths.get("../../../sample.daytrader7");
```
Comment (mrizzi) [inline rules-java-diva/tests/src/test/java/org/jboss/windup/rules/apps/diva/DivaTest.java:105]: This path must be changed sooner or later to have the test to be self-container without relying on external resources.

**Review by mrizzi — COMMENTED** [review]
Just a comment for a thing to be fixed.

File: rules-java-diva/tests/src/test/java/org/jboss/windup/rules/apps/diva/DivaTest.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            Path outputPath = Paths.get(FileUtils.getTempDirectory().toString(), "Windup")
+                    .resolve(UUID.randomUUID().toString());
+
+            WindupConfiguration windupConfiguration = new WindupConfiguration().setGraphContext(context)
+                    .setOptionValue(SourceModeOption.NAME, true).addInputPath(inputPath).setOutputDirectory(outputPath);
```
Comment (mrizzi) [inline rules-java-diva/tests/src/test/java/org/jboss/windup/rules/apps/diva/DivaTest.java:110]:
```suggestion
                    .setOptionValue(SourceModeOption.NAME, true).setOptionValue(EnableTransactionAnalysisOption.NAME, true).addInputPath(inputPath).setOutputDirectory(outputPath);
```

**Review by mrizzi — COMMENTED** [review]
I ran the instructions to test this and the `EnableTransactionAnalysisOption` was missing.

**Review by mrizzi — COMMENTED** [review]
I think the reports could look more like a "temporal" stack trace.
For example this report
Screenshot from 2022-02-08 18-43-10
could look more like this
```
com.ibm.websphere.samples.daytrader.web.prims.PingJDBCWrite.doPost 135
    com.ibm.websphere.samples.daytrader.web.prims.PingJDBCWrite.doGet 81
        com.ibm.websphere.samples.daytrader.direct.TradeDirect.updateQuotePriceVolumeInt 1321
            com.ibm.websphere.samples.daytrader.direct.TradeDirect.getQuoteForUpdate 819 
                "select * from quoteejb q where q.symbol=? For Update"
        com.ibm.websphere.samples.daytrader.direct.TradeDirect.updateQuotePriceVolumeInt 1336
            com.ibm.websphere.samples.daytrader.direct.TradeDirect.updateQuotePriceVolume 1364
                "update quoteejb set " + "price = ?, change1 = ?, volume = ? " + "where symbol = ?"
        com.ibm.websphere.samples.daytrader.direct.TradeDirect.updateQuotePriceVolumeInt 1337
            com.ibm.websphere.samples.daytrader.direct.TradeDirect.getQuote 801
                "select * from quoteejb q where q.symbol=?"
        com.ibm.websphere.samples.daytrader.direct.TradeDirect.updateQuotePriceVolumeInt 1339
            com.ibm.websphere.samples.daytrader.direct.TradeDirect.commit 1911
```
where indentation helps understanding which statements are executed in the same method/"level".

**Comment by akihikot** [comment]
The above changes. fix the broken indent issues for source highlighting.

**Comment by akihikot** [comment]
The changes in the PR are merged to #1427.  We close this PR.

**Pull request closed without merging**